src/Plant/dynamic/update_tree_with_time.jl

```diff
@@ -0,0 +1,75 @@
+using Parameters
+
+# function to update the tree with time, default at a Δt=1s using Wang-2020 scheme
+function update_tree_with_time(tree::StructTree, Δt::FT=FT(1.0); scheme::String="Wang2020", updating=false) where {FT}
```

Comment:
```suggestion
function update_tree_with_time!(tree::Tree, Δt::FT=FT(1.0); scheme::String="Wang2020", updating=false) where {FT}
```
Functions that change the contents of the arguments, by convention, append `!` to the function name.

Also, we don’t need to include `Struct` in the `Tree` type